crates/weaver_forge/src/extensions/otel.rs

```diff
@@ -277,15 +277,15 @@ pub(crate) fn is_stable(input: &Value) -> bool {
     false
 }
 
-/// Checks if the input value is an object with a field named "stability" that has the value
-/// "experimental". Otherwise, it returns false.
+/// Checks if the input value is an object with a field named "stability" that has any value
+/// other than "stable" or "deprecated". Otherwise, it returns false.
 #[must_use]
 pub(crate) fn is_experimental(input: &Value) -> bool {
     let result = input.get_attr("stability");
 
     if let Ok(stability) = result {
         if let Some(stability) = stability.as_str() {
-            return stability == "experimental";
+            return stability != "stable" && stability != "deprecated";
```

Comment:
I think we have "deprecated" in stability levels here only for backward compatibility.

In semantic conventions, `deprecated` is an extra property that can coexist with stability.

We deprecate experimental attributes and there are things that are experimental and deprecated. And there will be stable and deprecated things at the same time.

So I think we should do

```suggestion
            return stability != "stable";
```

@lquerel do you think we can deprecate or remove `Deprecated` stability enum member?

https://github.com/open-telemetry/weaver/blob/7b70d9f8ce975f66957c5544e649b543a07fdc25/crates/weaver_semconv/src/stability.rs#L32

Semconv doesn't use it since [v1.24.0](https://github.com/open-telemetry/semantic-conventions/blob/106f880ccbf26443f115ce1f48c236ec6a0b6f1b/CHANGELOG.md?plain=1#L432) and weaver works with semconv starting 1.26.0

Reply:
+1.  We should actually treat `stability="deprecated"` as stability="development" for the purpose of comparisons.  We made that decision back when we deprecated "deprecated" on stability level.

Reply:
> back when we deprecated "deprecated"

😹

Reply:
Seems like there's two things being debated here:
1. Backwards compatibility: How is compatibility defined? As someone relatively new to this it feels a bit like the tail is wagging the dog. The semantic conventions are versioned e.g. 1.24.0 and then the [schema definition](https://github.com/open-telemetry/weaver/blob/c7e27b243f09ada21e67d12226c03d8b0ca3ad76/schemas/semconv.schema.json) is updated. Perhaps the schema definition should be versioned and the semantic conventions declare the version they're using. I wrote about that [here](https://github.com/open-telemetry/weaver/issues/481#issuecomment-2507681351) too. We could then maintain up to, say, 3 prior versions for weaver backwards compatibility. This is particularly important for folks like me who define their own company-wide semantic conventions, we need to adhere to a schema version rather than figure it out from a release of the otel semconvs.
2. OTEP 232 compatibility: If we're trying to adhere to this standard then `deprecated` should be defined in the `stability` field. (Although arguably now `stability` should be renamed `maturity`). [Deprecated is defined in OTEP 232](https://github.com/open-telemetry/oteps/blob/main/text/0232-maturity-of-otel.md#deprecated). The `deprecated` field could become an optional `deprecation_message` instead.

But... I've not been around for all the history here so this is just my relatively newbie viewpoints. I'll wait for @lquerel  to respond to the question above re removing the deprecated variant and then make the changes.

Reply:
1. The question here is if folks start using the new weaver, do their workflows continue working or do they break.  Qualitativately this means:
  - Codegen continues working without change.
  - Rego Policies continues working without change.
  - weaver.yaml configuration continues working without change. (So JQ filter expressions work without change).
  
  Regarding supporting old/new schema versions that's an orthogonal but related concern to what I'm worried about here.  We *do* need to define a more formal "backwards compat" strategy with tests for which old versions we resolve.  However, for now, if we cut a new version we don't want existing usage to break, instead there should be a warning that becomes an error when passing `--future`.   For this PR I think it's acceptable for some breakage, as I don't see good alternatives otherwise.
  
  2. In Semantic conventions, we decided that `deprecated` should track the *previous stability level* of a component.  This is because we *allow* cleanup of non-stable deprecated fields but do not allow this for stable-then-deprecated fields.  This is an additional restriction on semantic conventions.  In practice, we are still specification compliant with how we annotate things due to our markdown templates.  Let's not mix specification compliance with architectural design of weaver.